src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -72,7 +72,7 @@ private static void LoadAppLocalIcu(string icuSuffixAndVersion)
 
         private static IntPtr LoadLibrary(string library, bool failOnLoadFailure)
         {
-            if (!NativeLibrary.TryLoad(library, typeof(object).Assembly, DllImportSearchPath.ApplicationDirectory, out IntPtr lib) && failOnLoadFailure)
+            if (!NativeLibrary.TryLoad(library, typeof(object).Assembly, DllImportSearchPath.ApplicationDirectory | DllImportSearchPath.System32, out IntPtr lib) && failOnLoadFailure)
             {
```

Comment:
Just wondering if this can change anything in the app-local loading? I mean if the ICU library is not included in the app folder and that will try to load it from the system32. are we just protected by only the library name? what would be the case when running on RS2 with app-local and the app not carrying the ICU libs? would we load the system ones at that time? did we test this scenario?

Reply:
Good question. The icu DLLs on RS2 don't have a version in the name (ex: icuuc.dll), so I wonder if the app does give a version, would it try to load a version-less DLL if it doesn't find them in the app?

Reply:
Also, I don't _think_ you can have an empty version number... but I'm not sure off-hand.
https://docs.microsoft.com/en-us/dotnet/standard/globalization-localization/globalization-icu#app-local-icu

If you can, then it might end up loading the icu dlls from System32 on RS2.

Reply:
It seems like we anyway try to load from system first as a fast path for Windows:

https://github.com/dotnet/runtime/blob/8d1d4ceef6a213a4a3f6e0db4747a4986c2b2e97/src/coreclr/src/vm/dllimport.cpp#L6247

Reply:
> Also, I don't think you can have an empty version number... but I'm not sure off-hand.

You can't, when we get the value from the runtime switch, we check that it is not null or empty: https://github.com/dotnet/runtime/blob/8d1d4ceef6a213a4a3f6e0db4747a4986c2b2e97/src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs#L34

Reply:
Thanks @safern!
In that case, if you can't have an empty suffix or version number, then I don't see how it would attempt to load the system32 icu dlls on RS2, as those ones have no version number on them at all.

Reply:
> It seems like we anyway try to load from system first as a fast path for Windows:
> 
> https://github.com/dotnet/runtime/blob/8d1d4ceef6a213a4a3f6e0db4747a4986c2b2e97/src/coreclr/src/vm/dllimport.cpp#L6247

Hmm, I wonder if maybe that was added or changed recently?

FWIW, I curiously didn't see any attempts to load the app-local ICU DLLs from System32 with a test app in ProcMon using preview6.

![image](https://user-images.githubusercontent.com/29107334/86194338-1764cf00-bb03-11ea-81f9-97e8751391fa.png)

Reply:
> Hmm, I wonder if maybe that was added or changed recently?

I could be wrong, maybe FEATURE_CORESYSTEM is not defined in this case. @jkotas when is that defined?

Reply:
`FEATURE_CORESYSTEM` is defined for CoreCLR. This path has been there for last ~10 years. Notice that it won't be taken for ICU. It is only taken only for the fake Windows API set .dlls. IIRC, it was added because there was something in the regular .dll loading path that choked on the API sets.

Reply:
Cool, thanks @jkotas